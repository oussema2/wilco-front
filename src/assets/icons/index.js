/* eslint-disable global-require */
export const aircraft = require( './aircraft.png' );
export const airport = require( './airport.png' );
export const arrowLeft = require( './arrowLeft.png' );
export const arrowRight = require( './arrowRight.png' );
export const award = require( './award.png' );
export const ban = require( './ban.png' );
export const bell = require( './bell.png' );
export const binoculars = require( './binoculars.png' );
export const brandSymbol = require( './brandSymbol.png' );
export const chevronLeft = require( './chevronLeft.png' );
export const chevronRight = require( './chevronRight.png' );
export const cloud = require( './cloud.png' );
export const certificate = require( './certificate.png' );
export const chatComment = require( './chatComment.png' );
export const chatNewMessage = require( './chatNewMessage.png' );
export const checkbox = require( './checkbox.png' );
export const checkboxUnselected = require( './checkboxUnselected.png' );
export const comment = require( './comment.png' );
export const cross = require( './cross.png' );
export const eye = require( './eye.png' );
export const eyeSlash = require( './eyeSlash.png' );
export const exclamationCircle = require( './exclamationCircle.png' );
export const exclamationTriangle = require( './exclamationTriangle.png' );
export const filter = require( './filter.png' );
export const flightInfoAircraft = require( './flightInfoAircraft.png' );
export const home = require( './home.png' );
export const horizontalLogo = require( './horizontalLogo.png' );
export const infoCircle = require( './infoCircle.png' );
export const logOut = require( './logOut.png' );
export const map = require( './map.png' );
export const meatballsMenu = require( './meatballsMenu.png' );
export const megaphone = require( './megaphone.png' );
export const onBoardingFirst = require( './onBoardingFirst.png' );
export const onBoardingSecond = require( './onBoardingSecond.png' );
export const onBoardingThird = require( './onBoardingThird.png' );
export const planeDeparture = require( './planeDeparture.png' );
export const plus = require( './plus.png' );
export const photoVideo = require( './photoVideo.png' );
export const privacyOnlyMe = require( './privacyOnlyMe.png' );
export const privacyPublic = require( './privacyPublic.png' );
export const remove = require( './remove.png' );
export const route = require( './route.png' );
export const rulerVertical = require( './rulerVertical.png' );
export const star = require( './star.png' );
export const starMyFeed = require( './starMyFeed.png' );
export const search = require( './search.png' );
export const settings = require( './settings.png' );
export const sliders = require( './sliders.png' );
export const speed = require( './speed.png' );
export const stopwatch = require( './stopwatch.png' );
export const thumbsUp = require( './thumbsUp.png' );
export const user = require( './user.png' );
export const users = require( './users.png' );
export const vector = require( './vector.png' );
