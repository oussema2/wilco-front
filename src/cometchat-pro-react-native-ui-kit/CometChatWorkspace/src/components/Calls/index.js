export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
