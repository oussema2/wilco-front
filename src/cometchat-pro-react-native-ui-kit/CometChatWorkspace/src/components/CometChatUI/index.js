/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
