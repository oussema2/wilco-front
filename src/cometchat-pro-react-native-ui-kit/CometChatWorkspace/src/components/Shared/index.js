export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
