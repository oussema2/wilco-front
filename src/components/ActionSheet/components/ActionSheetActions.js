import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
