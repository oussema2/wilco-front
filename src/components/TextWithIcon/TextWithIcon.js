import React from 'react';
import {
	Image, Text, View, ViewPropTypes
} from 'react-native';
import PropTypes from 'prop-types';

import { styles } from './TextWithIcon.styles';

const TextWithIcon = ( {
	testID, icon, children, style, imageStyle, numberOfLines
} ) => (
	<View testID={testID} style={[ styles.textWithIcon, style ]}>
		<View style={styles.iconView}>
			<Image style={[ styles.icon, imageStyle ]} source={icon} />
		</View>
		<Text style={styles.text} numberOfLines={numberOfLines}>
			{children}
		</Text>
	</View>
);

TextWithIcon.propTypes = {
	testID: PropTypes.string,
	icon: PropTypes.any.isRequired,
	children: PropTypes.string,
	style: ViewPropTypes.style,
	imageStyle: ViewPropTypes.style,
	numberOfLines: PropTypes.number
};

TextWithIcon.defaultProps = {
	testID: 'text-with-icon',
	children: '',
	style: null,
	imageStyle: null,
	numberOfLines: undefined
};

export default TextWithIcon;
