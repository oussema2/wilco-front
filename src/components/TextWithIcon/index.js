export { default as TextWithIcon } from './TextWithIcon';
