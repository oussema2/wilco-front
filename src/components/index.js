export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
