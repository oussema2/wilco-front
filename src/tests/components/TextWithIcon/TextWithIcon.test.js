import React from 'react';
import { render } from '@testing-library/react-native';
import { TextWithIcon } from '../../../components/TextWithIcon';
import { aircraft } from '../../../assets/icons';

describe( '<TextWithIcon />', () => {
	describe( 'with the default props', () => {
		it( 'renders the TextWithIcon component correctly', async () => {
			const component = render( <TextWithIcon icon={aircraft} /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with a text', () => {
		it( 'renders the TextWithIcon component correctly', async () => {
			const text = 'label';
			const component = render( <TextWithIcon icon={aircraft}>{text}</TextWithIcon> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with a numberOfLines', () => {
		it( 'renders the TextWithIcon component correctly', async () => {
			const component = render( <TextWithIcon
				icon={aircraft}
				numberOfLines={1}
			/>
			);

			expect( component ).toMatchSnapshot();
		} );
	} );
} );
