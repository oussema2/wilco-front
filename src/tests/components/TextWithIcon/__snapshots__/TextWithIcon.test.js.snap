// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextWithIcon /> with a numberOfLines renders the TextWithIcon component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      null,
    ]
  }
  testID="text-with-icon"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 28,
        "justifyContent": "center",
        "width": 28,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/aircraft.png",
        }
      }
      style={
        Array [
          Object {
            "height": 14,
            "tintColor": "#1F2E4E",
            "width": 20,
          },
          null,
        ]
      }
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      Object {
        "fontFamily": "NotoSansJP-Bold",
        "fontSize": 14,
        "fontWeight": undefined,
        "letterSpacing": 0,
        "lineHeight": 20,
        "marginLeft": 4,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`<TextWithIcon /> with a text renders the TextWithIcon component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      null,
    ]
  }
  testID="text-with-icon"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 28,
        "justifyContent": "center",
        "width": 28,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/aircraft.png",
        }
      }
      style={
        Array [
          Object {
            "height": 14,
            "tintColor": "#1F2E4E",
            "width": 20,
          },
          null,
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontFamily": "NotoSansJP-Bold",
        "fontSize": 14,
        "fontWeight": undefined,
        "letterSpacing": 0,
        "lineHeight": 20,
        "marginLeft": 4,
      }
    }
  >
    label
  </Text>
</View>
`;

exports[`<TextWithIcon /> with the default props renders the TextWithIcon component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      null,
    ]
  }
  testID="text-with-icon"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 28,
        "justifyContent": "center",
        "width": 28,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/aircraft.png",
        }
      }
      style={
        Array [
          Object {
            "height": 14,
            "tintColor": "#1F2E4E",
            "width": 20,
          },
          null,
        ]
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontFamily": "NotoSansJP-Bold",
        "fontSize": 14,
        "fontWeight": undefined,
        "letterSpacing": 0,
        "lineHeight": 20,
        "marginLeft": 4,
      }
    }
  >
    
  </Text>
</View>
`;
